pkg/migration/migrationmanager/manager.go

```diff
@@ -230,15 +243,17 @@ func (m *Manager) Migrate(
 // until the cluster is stable.
 func bumpClusterVersion(
 	ctx context.Context, c migration.Cluster, clusterVersion clusterversion.ClusterVersion,
-) error {
+) (error, bool) {
 	req := &serverpb.BumpClusterVersionRequest{ClusterVersion: &clusterVersion}
 	op := fmt.Sprintf("bump-cluster-version=%s", req.ClusterVersion.PrettyPrint())
+	anyVersionBumped := false
```

Comment:
I'm not sure if this is sound, and I'm not sure that using this return value to decide whether or not to update system.settings is sound either. The guarantees we want to provide at the level of `(*Manager) Migrate` is that it should be completely idempotent. With the hook passed in, I think we want to instead check that the version stored in system.settings is less than the version we're trying to set it to.